api/src/DataPersister/ResetPasswordDataPersister.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\DataPersister;
+
+use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
+use App\DataPersister\Util\DataPersisterObservable;
+use App\DTO\ResetPassword;
+use App\Repository\UserRepository;
+use App\Service\MailService;
+use App\Util\IdGenerator;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use Symfony\Component\PasswordHasher\Hasher\PasswordHasherFactoryInterface;
+use Symfony\Component\PasswordHasher\PasswordHasherInterface;
+
+class ResetPasswordDataPersister implements ContextAwareDataPersisterInterface {
+    public function __construct(
+        private DataPersisterObservable $dataPersisterObservable,
+        private EntityManagerInterface $em,
+        private UserRepository $userRepository,
+        private PasswordHasherFactoryInterface $pwHasherFactory,
+        private MailService $mailService
+    ) {
+    }
+
+    public function supports($data, array $context = []): bool {
+        return $data instanceof ResetPassword;
+    }
+
+    public function persist($data, array $context = []): void {
+        $observable = $this->dataPersisterObservable
+            ->onBeforeCreate(fn ($data) => $this->beforeCreate($data))
+            ->onBeforeUpdate(fn ($data) => $this->beforeUpdate($data))
+        ;
+
+        $data = $observable->persist($data, $context);
+        $this->em->flush();
```

Comment:
is the flush necessary?

Reply:
Da ich hier nur das Interface `ContextAwareDataPersisterInterface` implementiere - aber nicht von `AbstractDataPersister` ableite - ja